package/MDAnalysis/analysis/base.py

```diff
@@ -24,61 +24,116 @@
 """
 from six.moves import range
 
-import numpy as np
 import logging
 from MDAnalysis.lib.log import ProgressMeter
 
 logger = logging.getLogger(__name__)
 
 
 class AnalysisBase(object):
-    """Base class for defining multi frame analysis
-
-    The analysis base class is designed as a template for creating
-    multiframe analysis.
-
-    The class implements the following methods:
-
-    _setup_frames(trajectory, start=None, stop=None, step=None)
-      Pass a Reader object and define the desired iteration pattern
-      through the trajectory
-
-    run
-      The user facing run method.  Calls the analysis methods
-      defined below
-
-    Your analysis can implement the following methods, which are
-    called from run:
-
-    _prepare
-      Called before iteration on the trajectory has begun.
-      Data structures can be set up at this time, however most
-      error checking should be done in the __init__
-
-    _single_frame
-      Called after the trajectory is moved onto each new frame.
-
-    _conclude
-      Called once iteration on the trajectory is finished.
-      Apply normalisation and averaging to results here.
+    """Base class for defining multi frame analysis, it is designed as a
+    template for creating multiframe analysis. This class will automatically
+    take care of setting up the trajectory reader for iterating and offers to
+    show a progress meter.
+
+    To define a new Analysis, `AnalysisBase` needs to be subclassed
+    `_single_frame` must be defined. It is also possible to define
+    `_prepare` and `_conclude` for pre and post processing. See the example
+    below.
+
+    .. code-block:: python
```

Comment:
> On 12 Jul, 2016, at 02:49, Max Linke <notifications@github.com> wrote:
> 
> how can we easily link to this? I thought it would be enough to make an See Also mention in the other classes docs.

I suppose that works. Just remark somewhere above that this piece of code *is* the definition of what the Analysis API looks like at the moment.

I was thinking about a document similar to the Trajectory API but to be honest, this PR is not really the place for that.

> I can make some more comments about the individual functions that can be overwritten in a child class. would that be better for you?

Anything that helps developers :-) – but I’d say we leave all of this for another PR, say, “Writing Analysis classes docs” (or have we already got something like that?).